cql3/query_processor.cc

```diff
@@ -1047,6 +1047,14 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
         } else {
             co_await remote_.get().mm.announce<service::schema_change>(std::move(m), std::move(*guard), description);
         }
+        // Internal queries don't trigger auto-grant. Statements which don't require
+        // auto-grant use default nop grant_permissions_to_creator implementation.
+        // Additional implicit assumption is that prepare_schema_mutations returns
+        // empty mutations vector when resource doesn't need to be created (e.g. already exists).
+        auto& client_state = state.get_client_state();
+        if (!client_state.is_internal()) {
+            co_await stmt.grant_permissions_to_creator(client_state);
```

Comment:
Out of scope, but we should make this part of the transaction.

Reply:
Or maybe it already is? via client_state?

Reply:
it's in review, https://github.com/scylladb/scylladb/pull/17910 and potential followups

Reply:
ok